### PR TITLE
rate limit perma fix

### LINKED ISSUE #505: Still polling too soon. (Too many Request)
it is still getting "Too many requests" error (400), after every 2 requests.

### PR DESCRIPTION
**Pull request by helloworld111gh**
fixes #505, also will auto adjust if rate limit increases again
i didn't test it, but it should work

### CODE REVIEW

**Comment by JPierreS** [comment]
Can you comment on line 48 of the code ? Adding of 5 seconds ?
How about a method to get the exact rate and match it 1:1  to avoid setting it
too high above the actual limit.

**Comment by helloworld111gh** [comment]
since rainforest doesn't tell you the current rate limit in the api, the pr
will just guess and keeps adding 5 seconds until it stops getting rate
limited

On 11 September 2016 at 21:58, Jean-Pierre Sabbagh <notifications@github.com

**Comment by sweeneyapps** [comment]
well, ur code won't work.. but I can give u some pointers..

File: src/reducers/polling.js [hunk 1]
```diff
@@ -45,8 +45,8 @@ const captchaRequired = (state) => (
 );
 
 const rateLimitExceeded = (state) => {
-  let newState = (state.get('interval') + 1000);
-  return state.set('interval', newState); // keep adding 1 second till we are not rate limited
+  const newState = (state.get('interval') + (3 * 1000));
```
Comment (sweeneyapps) [inline src/reducers/polling.js]: just realized my mistake..   newState wouldn't sound right.     how about  newInterval  or something that makes sense.

File: src/constants.js [hunk 2]
```diff
@@ -27,7 +28,7 @@ export const colors = deepFreeze({
 
 export const REDUCERS = deepFreeze(['worker', 'socket', 'plugin', 'polling']);
 
-export const DEFAULT_POLLING_INTERVAL = 20 * 1000;
+export const DEFAULT_POLLING_INTERVAL = 28 * 1000;
```
Comment (shosti) [inline src/constants.js]: I think this should be 30 for now.

**Comment by shosti** [comment]
Approach looks sane. Can you:

1. Merge in the latest develop (will add CI with travis)
2. Run `npm run lint` and fix style problems
3. Write a quick test for the reducer

**Comment by sweeneyapps** [comment]
waiting on merging my code: https://github.com/helloworld111gh/tester-chrome-extension/pull/2

**Pull request merged**